source/extensions/load_balancing_policies/maglev/maglev_lb.cc

```diff
@@ -262,12 +262,13 @@ HostConstSharedPtr OriginalMaglevTable::chooseHost(uint64_t hash, uint32_t attem
     hash ^= ~0ULL - attempt + 1;
   }
 
-  return table_[hash % table_size_];
+  HostConstSharedPtr host = table_[hash % table_size_];
+  return {std::move(host)};
```

Comment:
Since the `HostSelectionResponse` constructor doesn't require a `&&`, I don't think move is required. Storing the host in a local variable makes a copy, so doing the `move` on the variable is still a single copy. I think it would probably just be cleaner to do:

` return {table_[hash % table_size_]};`

WDYT? (Here and below)

Reply:
yeah some iteration was having inference issues but all good now - cleaned up here and below.